src/inbox/tx.rs

```diff
@@ -0,0 +1,437 @@
+use super::*;
+use crate::inbox::tx::private::RefCounterInner;
+use crate::Disconnected;
+use event_listener::EventListener;
+use futures_core::FusedFuture;
+use futures_util::FutureExt;
+use std::fmt::{Debug, Formatter};
+use std::future::Future;
+use std::mem;
+use std::pin::Pin;
+use std::sync::{atomic, Arc};
+use std::task::{Context, Poll, Waker};
+
+pub struct Sender<A, Rc: TxRefCounter> {
+    pub(super) inner: Arc<Chan<A>>,
+    pub(super) rc: Rc,
+}
+
+impl<A> Sender<A, TxStrong> {
+    pub fn new(inner: Arc<Chan<A>>) -> Self {
+        let rc = TxStrong(());
+        rc.increment(&inner);
+
+        Sender { inner, rc }
+    }
+}
+
+impl<Rc: TxRefCounter, A> Sender<A, Rc> {
```

Comment:
Would it be okay to put `try_send` first? I think it's probably the most likely suspect when you go into the code for `Sender`, so having it near the top could be nice

Reply:
Fair. I tend to put public APIs at the top because they are usually the functions you know about which helps to build up a mental model on what code is called where. Up to you!